Pacemaker/pcs_cluster_36_nodes_feasibility/scale-cluster-upto-36-nodes.md

```diff
@@ -0,0 +1,155 @@
+
+  # Now unstandby and start nodes
+  $ pcs -f cortx-pcs.xml cluster unstandby srvnode-1
+  $ pcs -f cortx-pcs.xml cluster start srvnode-2
+  $ pcs -f cortx-pcs.xml cluster start srvnode-3
+
+  # Verify and push file
+  $ pcs cluster verify -V cortx-pcs.xml
+  $ pcs cluster cib-push cortx-pcs.xml
+  ```
+
+- System Utilization (ram/cpu/network)
+  - DC node will use high utilization as it will act as leader and responsible for action.
+  - Utilization depend on no. node and resource going to affect.
+  - Standby 3 node will have less utilization but standby all node will have high utilization.
+
+- s3 and motr like service.
+  - use generic resource to create multiple clone on same node.
+  - we can modify clone size while scaling cluster.
+    ```
+    # Example
+    $ pcs resource create motr ocf:seagate:s3server service=motr unique_clone=true clone clone-max=10 clone-node-max=2 globally-unique=true
+
+    # After scaling change clone size
+    $ pcs resource update motr-clone meta clone-max=20 clone-node-max=2
+    ```
+
+Outcome:
+
+## 1. Add nodes up to 36 verify corosync limit
+- Able to create 36 nodes cluster with 32 normal nodes and 4 remote nodes.
+- We created scripts to create 36 nodes cluster and already shared in the scripts dir.
+
+## 2. Configure the dummy resources with the dependency
+- Configured around 618 resources with the dependecy in the cluster and observed that pcs cluster became unstable.
+- We observed from corosync logs, PCMK_ipc_buffer has exceeded buffer limit and also saw some bad file descriptor related errors.
+- We tunned some parameters to make cluster stable. Please find details in tunning section.
+
+Check PCS Output after configuring the resources (`check pcs_status.txt`)
+
+## 3. Cluster tunning details
+
+Below issue observed after scaling cluster and configuring resources
+- PCMK_ipc_buffer exceeded
+- Bad file descriptor related errors
+
+**Tunning performed**
+1. IPC buffer limit
+- Increased IPC buffer limit (PCMK_ipc_buffer=19725604)
+- IPC buffer is the size of message used to corosync for heartbeat communication.
+
+2. Open file limit
+- Increased open file limit to 1048575 (ulimit -n 1048575)
+- We observed bad file descriptor erros in the corosync logs and did above change to fix issue.
+
+3. Batch-limit
+- Batch-limit is the maximum number of actions that the cluster may execute in parallel across all nodes. The "correct" value will depend on the speed and load of your network and cluster nodes. If zero, the cluster will impose a dynamically calculated limit only when any node has high load.
+- We tried 25 fixed batch-limit instead of dynamic limit and observed that network increased upto 345 Mbits/s and due to high CPU usage DC node got also changes. We also observed that some resources got time-out during unstandby nodes operation.
+- We reverted batch-limit to zero and network load back to normal. So, We concluded batch-limit zero is suitable for large cluster.
```

Comment:
Do we have all of such recommendations documented at one place?

Reply:
Updated link to the doc which has all the cluster options. We found only one cluster option (cluster-ipc-limit) which is useful for the large cluster.

Reply:
Sure